cpp/oneapi/dal/io/detail/load_graph.hpp

```diff
@@ -53,126 +54,127 @@ edge_list<std::int32_t> load_edge_list(const std::string &name) {
     if (rows_cv == nullptr) {
         throw bad_alloc();
     }
 
     vertex_t total_sum_degrees = 0;
     rows_cv[0].set(total_sum_degrees);
-
-    for (vertex_t i = 0; i < _unf_vertex_count; ++i) {
+    for (vertex_t i = 0; i < n_vertex; ++i) {
         total_sum_degrees += degrees_cv[i].get();
         rows_cv[i + 1].set(total_sum_degrees);
     }
-    delete degrees_vec;
+    allocator.deallocate((char *)degrees_vec_void, n_vertex * (sizeof(atomic_t) / sizeof(char)));
+
+    void *_unf_vert_neighs_vec_void =
+        (void *)allocator.allocate((rows_cv[n_vertex].get()) * (sizeof(vertex_t) / sizeof(char)));
+    vertex_t *_unf_vert_neighs_arr =
+        new (_unf_vert_neighs_vec_void) vertex_t[rows_cv[n_vertex].get()];
 
-    vector_vertex_t _unf_vert_neighs_vec(rows_cv[_unf_vertex_count].get());
-    vector_edge_t _unf_edge_offset_vec(_unf_vertex_count + 1);
-    auto _unf_edge_offset_arr = _unf_edge_offset_vec.data();
-    auto _unf_vert_neighs_arr = _unf_vert_neighs_vec.data();
+    void *_unf_edge_offset_arr_void =
+        (void *)allocator.allocate((n_vertex + 1) * (sizeof(edge_t) / sizeof(char)));
+    edge_t *_unf_edge_offset_arr = new (_unf_edge_offset_arr_void) edge_t[n_vertex + 1];
 
-    threader_for(_unf_vertex_count + 1, _unf_vertex_count + 1, [&](vertex_t n) {
+    threader_for(n_vertex + 1, n_vertex + 1, [&](vertex_t n) {
         _unf_edge_offset_arr[n] = rows_cv[n].get();
     });
 
     threader_for(edges.size(), edges.size(), [&](vertex_t u) {
         _unf_vert_neighs_arr[rows_cv[edges[u].first].inc() - 1] = edges[u].second;
         _unf_vert_neighs_arr[rows_cv[edges[u].second].inc() - 1] = edges[u].first;
     });
-    delete rows_vec;
+    allocator.deallocate((char *)rows_vec_void, (n_vertex + 1) * (sizeof(atomic_t) / sizeof(char)));
 
-    //removing self-loops,  multiple edges from graph, and make neighbors in CSR sorted
+    layout->_degrees = std::move(vector_vertex_t(n_vertex));
+    auto _degrees_data = layout->_degrees.data();
 
-    layout->_vertex_count = _unf_vertex_count;
-
-    layout->_degrees = std::move(vector_vertex_t(layout->_vertex_count));
+    //removing self-loops,  multiple edges from graph, and make neighbors in CSR sorted
+    threader_for(n_vertex, n_vertex, [&](vertex_t u) {
+        auto start_p = _unf_vert_neighs_arr + _unf_edge_offset_arr[u];
+        auto end_p = _unf_vert_neighs_arr + _unf_edge_offset_arr[u + 1];
 
-    threader_for(_unf_vertex_count, _unf_vertex_count, [&](vertex_t u) {
-        auto start_p = _unf_vert_neighs_vec.begin() + _unf_edge_offset_vec[u];
-        auto end_p = _unf_vert_neighs_vec.begin() + _unf_edge_offset_vec[u + 1];
         std::sort(start_p, end_p);
+
         auto neighs_u_new_end = std::unique(start_p, end_p);
         neighs_u_new_end = std::remove(start_p, neighs_u_new_end, u);
-        layout->_degrees[u] = (vertex_t)std::distance(start_p, neighs_u_new_end);
+        _degrees_data[u] = (vertex_t)std::distance(start_p, neighs_u_new_end);
```

Comment:
Are you sure that your high integration with std does not make unwanted dependencies for oneDAL?

Reply:
It's only header file code.